### PR TITLE
fix: add ServiceAccount, Role and RoleBinding for fuseml-core

### LINKED ISSUE #88: fuseml-core installation: solve the permissions for creating the pipelines
Tekton uses `staging-triggers-admin` service account to create the pipelines, and fuseml-core deployment as is created now does not have the rights for relevant role.

We either need to 
- create fuseml-core with the same service account (and same namespace `fuseml-workflow`)
- create separate serviceAccount for fuseml-core and add it to clusterrolebinding `staging-triggers-admin-binding` so that it shares the same privilegies

### PR DESCRIPTION
**Pull request by jsuchome**
This is needed so that fuseml-core has permissions to create
tekton pipelines.

Closes #88.

### CODE REVIEW

File: embedded-files/fuseml-core-deployment.yaml [hunk 1]
```diff
@@ -1,4 +1,37 @@
...
+metadata:
+  name: fuseml-core-tekton
+  namespace: fuseml-workloads
+rules:
+- apiGroups:
+  - tekton.dev
+  resources:
+  - pipelines
+  verbs:
+  - create
```
Comment (flaviodsr) [inline embedded-files/fuseml-core-deployment.yaml:19]: Will that be enough? fuseml also creates `TriggerTemplate`, `TriggerBinding` and `EventListener` from Tekton triggers.

There is also a delete/get operation to be implemented (that could also be added later when we do have it).

**Review by flaviodsr — CHANGES_REQUESTED** [review]

File: embedded-files/fuseml-core-deployment.yaml (hunk 1 above)
Comment (jsuchome) [reply embedded-files/fuseml-core-deployment.yaml:19]: You are right of course

File: embedded-files/fuseml-core-deployment.yaml [hunk 2]
```diff
@@ -1,4 +1,52 @@
...
+  - get
+  - delete
+- apiGroups:
+  - triggers.tekton.dev
+  resources:
+  - eventlisteners
+  - triggerbindings
+  - triggertemplates
+  - triggers
```
Comment (flaviodsr) [inline embedded-files/fuseml-core-deployment.yaml:29]: FuseML does not perform any operation on triggers

File: embedded-files/fuseml-core-deployment.yaml [hunk 3]
```diff
@@ -1,4 +1,52 @@
...
+  verbs:
+  - get
+  - list
+  - watch
```
Comment (flaviodsr) [inline embedded-files/fuseml-core-deployment.yaml:33]: FuseML also does not perform `watch`. It does `delete` though...

**Review by flaviodsr — CHANGES_REQUESTED** [review]

File: embedded-files/fuseml-core-deployment.yaml (hunk 2 above)
Comment (jsuchome) [reply embedded-files/fuseml-core-deployment.yaml]: done

File: embedded-files/fuseml-core-deployment.yaml (hunk 3 above)
Comment (jsuchome) [reply embedded-files/fuseml-core-deployment.yaml]: done

File: embedded-files/fuseml-core-deployment.yaml [hunk 4]
```diff
@@ -1,4 +1,51 @@
...
+  name: fuseml-core-tekton
+  namespace: fuseml-workloads
+rules:
+- apiGroups:
+  - tekton.dev
+  resources:
+  - pipelines
+  - pipelineruns
+  verbs:
+  - create
```
Comment (flaviodsr) [inline embedded-files/fuseml-core-deployment.yaml:20]: I am sorry but after https://github.com/fuseml/fuseml-core/pull/34 merged, FuseML also needs permission to list pipelineruns.

**Review by flaviodsr — CHANGES_REQUESTED** [review]

File: embedded-files/fuseml-core-deployment.yaml (hunk 4 above)
Comment (jsuchome) [reply embedded-files/fuseml-core-deployment.yaml:20]: fixed

**Review by flaviodsr — APPROVED** [review]

**Pull request merged**